### PR TITLE
WRO-653: Create Tests(editableScroller).

### PR DESCRIPTION
**Pull request by dongsuwon**
Enact-DCO-1.0-Signed-off-by: Dongsu Won (dongsu.won@lgepartner.com)

### Checklist
* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
- Need to created screenshot tests for editableScroller.
- Need to created ui tests for editableScroller.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Create screenshot tests for editableScroller.
- Create ui tests for editableScroller.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]:# (Related issues, references)
WRO-653
QWT-4513

### CODE REVIEW

File: tests/screenshot/apps/components/Scroller.js [hunk 1]
```diff
@@ -1,12 +1,86 @@
...
 import Scroller from '../../../../Scroller';
+import ri from '@enact/ui/resolution';
 
 import css from './Scroller.module.less';
 
+const imageItems = [];
+const defaultdataSize = 3;
+
+const renderImageItem = (dataSize) => {
+
```
Comment (jeonghee27) [inline tests/screenshot/apps/components/Scroller.js:13]:
```suggestion

```

File: tests/screenshot/apps/components/Scroller.module.less [hunk 2]
```diff
@@ -4,4 +4,55 @@
 	.thumb {
 		background-color: orangered;
 	}
+
+
```
Comment (jeonghee27) [inline tests/screenshot/apps/components/Scroller.module.less:8]:
```suggestion
```

File: tests/screenshot/apps/components/Scroller.module.less [hunk 3]
```diff
@@ -4,4 +4,55 @@
...
+			left: -30px;
+		}
+
+		&::after {
+			content: '\EFFF4';
+			.arrow();
+			right: -30px;
+		}
+	}
+}
```
Comment (jeonghee27) [inline tests/screenshot/apps/components/Scroller.module.less:58]:
```suggestion
}

```

File: tests/screenshot/apps/components/Scroller.js [hunk 4]
```diff
@@ -1,12 +1,86 @@
...
+						>
+							{`Image ${item.index}`}
+						</ImageItem>
+					</div>
+				);
+			})
+		}
+	</Scroller>,
+	<Scroller
+		direction={'horizontal'}
```
Comment (jeonghee27) [inline tests/screenshot/apps/components/Scroller.js:59]: direction="horizontal"
```suggestion
		direction="horizontal"
```

File: tests/screenshot/apps/components/Scroller.js [hunk 5]
```diff
@@ -1,12 +1,86 @@
...
+
+const renderImageItem = (dataSize) => {
+
+	imageItems.length = 0;
+
+	for (let i = 0; i < dataSize; i++) {
+		const index = i;
+		const source = {img};
+		imageItems.push({source, index});
+	}
```
Comment (jeonghee27) [inline tests/screenshot/apps/components/Scroller.js:20]: If we use same "img", we don't need imageItems obj

Please think more to make simple code

imageItems = Array.from({length: dataSize}, (v, i) => i); ? 
I'm not sure.

File: tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+const OptionsContainer = SpotlightContainerDecorator({leaveFor: {down: '#left'}}, 'div');
+
+// NOTE: Forcing pointer mode off so we can be sure that regardless of webOS pointer mode the app
+// runs the same way
+spotlight.setPointerMode(false);
+
+const imageItems = [];
+
+const updateDataSize = (dataSize) => {
+
```
Comment (jeonghee27) [inline tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js:24]:
```suggestion
```

File: tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+				<Cell component={OptionsContainer} shrink>
+					<Button {...buttonDefaultProps} id="nativeScroll" onClick={this.onToggle}>{scrollMode}</Button>
+					<Button {...buttonDefaultProps} id="editableCentered" onClick={this.onToggle}>EditableCentered</Button>
+					<InputField id="numItems" defaultValue={numItems} type="number" onChange={this.onChangeNumItems} size="small" style={inputStyle} />
+				</Cell >
+				<Cell component={ScrollerContainer}>
+					<Scroller
+						dataSize={numItems}
+						direction={'horizontal'}
+						horizontalScrollbar={'auto'}
```
Comment (jeonghee27) [inline tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js:99]:
```suggestion
						direction="horizontal"
						horizontalScrollbar="auto"
```

File: tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+						key={nativeScroll ? 'native' : 'translate'}
+						verticalScrollbar={'auto'}
```
Comment (jeonghee27) [inline tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js:101]:
```suggestion
						verticalScrollbar="auto"
```

File: tests/ui/specs/Scroller/EditableScroller/Scroller-QWT-4513-specs.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
+const ScrollerPage = require('../ScrollerPage');
+const {expectFocusedItem} = require('../Scroller-utils');
+
+describe('Editable Scroller', function () {
+	beforeEach(async function () {
+		await ScrollerPage.open('EditableItem');
+	});
+
+	it('Should change item position with editableCentered', async function () {
```
Comment (jeonghee27) [inline tests/ui/specs/Scroller/EditableScroller/Scroller-QWT-4513-specs.js:9]: QWT-4513 also check center position .
([Enact QA Sampler] sandstone/Scroller>with editable items: Items horizontally center align in Scroller with 'editableCentered')

Could you please add another tc ("it")

File: tests/ui/specs/Scroller/EditableScroller/Scroller-QWT-4514-specs.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
+const ScrollerPage = require('../ScrollerPage');
+const {expectFocusedItem} = require('../Scroller-utils');
+
+describe('Editable Scroller', function () {
+	beforeEach(async function () {
+		await ScrollerPage.open('EditableItem');
+	});
+
+	it('Should change item position with 5-way mode', async function () {
```
Comment (jeonghee27) [inline tests/ui/specs/Scroller/EditableScroller/Scroller-QWT-4514-specs.js:9]: 👍🏻

File: tests/screenshot/apps/components/Scroller.module.less (hunk 2 above)
Comment (dongsuwon) [reply tests/screenshot/apps/components/Scroller.module.less]: Suggested change.

File: tests/screenshot/apps/components/Scroller.module.less (hunk 3 above)
Comment (dongsuwon) [reply tests/screenshot/apps/components/Scroller.module.less]: Suggested change.

File: tests/screenshot/apps/components/Scroller.js (hunk 4 above)
Comment (dongsuwon) [reply tests/screenshot/apps/components/Scroller.js]: Suggested change.

File: tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js (hunk 6 above)
Comment (dongsuwon) [reply tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js]: Suggested change.

File: tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js (hunk 7 above)
Comment (dongsuwon) [reply tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js]: Suggested change.

File: tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js (hunk 8 above)
Comment (dongsuwon) [reply tests/ui/apps/Scroller/EditableItem/ScrollerEditableItem-View.js]: Suggested change.

File: tests/ui/specs/Scroller/EditableScroller/Scroller-QWT-4513-specs.js (hunk 9 above)
Comment (dongsuwon) [reply tests/ui/specs/Scroller/EditableScroller/Scroller-QWT-4513-specs.js:9]: Step 3~ Step4 are worked in screenshot tests. I added comment about QWT-4513. Please check this.

File: tests/ui/specs/Scroller/EditableScroller/Scroller-QWT-4514-specs.js (hunk 10 above)
Comment (dongsuwon) [reply tests/ui/specs/Scroller/EditableScroller/Scroller-QWT-4514-specs.js:9]: Thank you!

File: tests/screenshot/apps/components/Scroller.js (hunk 5 above)
Comment (dongsuwon) [reply tests/screenshot/apps/components/Scroller.js]: Suggested Change.

**Review by jeonghee27 — APPROVED** [review]
LGTM

**Pull request merged**